docs/operational_guide/availability_consistency_durability.md

```diff
@@ -0,0 +1,126 @@
+To address as many use cases as possible, M3DB can be tuned to achieve the desired balance between performance, availability, durability, and consistency.
+
+Generally speaking, [the default and example configuration for M3DB](https://github.com/m3db/m3/tree/master/src/dbnode/config) favors performance and availability, as that is well-suited for M3DB's most common metrics and Observability use cases. To instead favor consistency and durability, consider tuning values as described in (TODO).
+Database operators who are using M3DB for workloads that require stricter consistency and durability guarantees should consider tuning the default configuration to better suite their use case.
+The rest of this document describes the various configuration options that are available to M3DB operators to make those tradeoffs.
+
+While reading the rest of this document, we recommend referring to [the default configuration file](https://github.com/m3db/m3/blob/master/src/dbnode/config/m3dbnode-all-config.yml) (which has every possible configuration value set) to see how the described values fit into M3DB's configuration as a whole.
+
+## Tuning for Performance and Availability
+
+### Client Write and Read consistency
+
+We recommend running the client with `writeConsistencyLevel` set to `majority` and `readConsistencyLevel` set to `unstrict_majority`.
+This means that all write must be acknowledged by a quorums of nodes in order to be considered succesful, and that reads will attempt to achieve quorum, but will return the data from a single node if they are unable to achieve quorum.
+You can read about the consistency levels in more detail in [the Consistency Levels section](../m3db/architecture/consistencylevels.md)
+
+### Commitlog Configuration
+
+We recommend running M3DB with an asynchronous commitlog.
+This means that writes will be reported as successful by the client, though the data may not have been flushed to disk yet.
+
+For example, consider the default configuration:
+
+```
+commitlog:
+  flushMaxBytes: 524288
+  flushEvery: 1s
+  queue:
+    calculationType: fixed
+    size: 2097152
+```
+
+This configuration states that the commitlog should be flushed whenever either of the following is true:
+
+1. 524288 or more bytes have been written since the last time M3DB flushed the commitlog.
+2. One or more seconds has elapsed since the last time M3DB flushed the commitlog.
+
+In addition, the configuration also states that M3DB should allow up to `2097152` writes to be buffered in the commitlog queue before the database node will begin rejecting incoming writes so it can attempt to drain the queue and catch up. Increasing the size of this queue can often increase the write throughput of an M3DB node at the cost of potentially losing more data if the node experiences a sudden failure like a hard crash or power loss.
```

Comment:
> drain the queue and catch up

i don't think (but am potentially just not seeing) that we've documented/diagrammed the architectural details to explain how the queue works (maybe that'd be overkill), but if we ever add that, this should probably link to it.

Reply:
Yeah unfortunately we don't have that right now

Reply:
i didn't think so - just a note for the future, then. :)